src/functions.js

```diff
@@ -121,7 +120,30 @@ var _ = Mavo.Functions = {
 			return n && n > 1? [arr] : arr;
 		}
 
-		return n && n > 1? arr.slice(0, n) : arr[0];
+		if (n && n > 1) {
+			tempArr = [];
+			var i = 0;
+
+			while (i<arr.length && tempArr.length<n) {
+
+				if (arr[i]) {
```

Comment:
This will work with actual `null`, but not with the `null` we use in expressions, which is an object, and truthy just like every object (fun fact: even `new Boolean(false)` is truthy in JS).
Also, it will not add `0` and empty strings to the array, which you **do** want.
I'd suggest `Mavo.value(arr[i]) !== null`.